js/ftx.js

```diff
@@ -1879,6 +1882,68 @@ module.exports = class ftx extends Exchange {
         return this.parseTrades (trades, market, since, limit);
     }
 
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const request = {
+            'coin': currency['id'],
+            'source': fromAccount,
+            'destination': toAccount,
+            'size': amount,
+        };
+        const response = await this.privatePostSubaccountsTransfer (this.extend (request, params));
+        //
+        //     {
+        //         success: true,
+        //         result: {
+        //             id: '31222278',
+        //             coin: 'USDT',
+        //             size: '1.0',
+        //             time: '2022-04-01T11:18:27.194188+00:00',
+        //             notes: 'Transfer from main account to testSubaccount',
+        //             status: 'complete'
+        //         }
+        //     }
+        //
+        const result = this.safeValue (response, 'result', {});
+        return this.parseTransfer (result, currency);
+    }
+
+    parseTransfer (transfer, currency = undefined) {
+        //
+        //     {
+        //         id: '31222278',
+        //         coin: 'USDT',
+        //         size: '1.0',
+        //         time: '2022-04-01T11:18:27.194188+00:00',
+        //         notes: 'Transfer from main account to testSubaccount',
+        //         status: 'complete'
+        //     }
+        //
+        const currencyId = this.safeString (transfer, 'coin');
+        const notes = this.safeString (transfer, 'notes', '');
+        const status = this.safeString (transfer, 'status');
+        const statuses = {
+            'complete': 'ok',
+        };
+        const fromAccountStartIndex = notes.indexOf ('Transfer from ') + 14;
+        const fromAccountEndIndex = notes.indexOf (' to ');
+        const fromAccount = notes.slice (fromAccountStartIndex, fromAccountEndIndex);
+        const toAccountStartIndex = notes.indexOf (' to ') + 4;
```

Comment:
Let's rewrite lines 1929 and 1932 for `this.sum()`, otherwise we violate this rule:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings

Also, we could remove the duplication from line 1932:

```JavaScript
const fromAccountStartIndex = 14;
const fromAccountEndIndex = notes.indexOf (' to ');
const fromAccount = notes.slice (fromAccountStartIndex, fromAccountEndIndex);
const toAccountStartIndex = this.sum (fromAccountStartIndex, 4);
```

Alternatively, we can do it this way (i think it may be cleaner and a bit more readable):

```JavaScript
const fromTo = notes.replace ('Transfer from ', '');
const parts = fromTo.split (' to ');
const fromAccount = this.safeString (parts, 0);
const toAccount = this.safeString (parts, 1);
```

Reply:
I like it.
I also moved the status to `parseTransferStatus`